### PR TITLE
Runners more resilient to docker login failure

### PR DESCRIPTION
**Pull request by potiuk**
Login to docker registry is now done in PreExec and in case it
fails, it also fails the whole service (leading to subsequent
service restart).

Also added `set -eux -o pipefail` to be better protected against
any silent failures.

### CODE REVIEW

File: cloud-init.yml [hunk 1]
```diff
@@ -145,7 +149,7 @@ write_files:
   - path: /usr/local/bin/stop-runner-if-no-job.sh
     content: |
       #!/bin/bash
-      set -u
+      set -exu -o pipefail
```
Comment (ashb) [inline cloud-init.yml:152]:
```suggestion
      set -eu -o pipefail
```

File: cloud-init.yml [hunk 2]
```diff
@@ -123,10 +121,16 @@ write_files:
   - path: /usr/local/sbin/runner-cleanup-workdir.sh
     content: |
       #!/bin/bash
+      set -exu -o pipefail
```
Comment (ashb) [inline cloud-init.yml:124]:
```suggestion
      set -eu -o pipefail
```

File: cloud-init.yml [hunk 3]
```diff
@@ -195,7 +199,7 @@ write_files:
     permissions: '0775'
     content: |
       #!/bin/bash
-
+      set -exu -o pipefail
```
Comment (ashb) [inline cloud-init.yml:202]:
```suggestion
      set -eu -o pipefail
```
This is a cron job, so shouldn't produce needless output.

**Comment by potiuk** [comment]
Sure - no problem for me. I just do it by default on such scipts to know what's going on when there are problems, but no strong feelings about it.

**Review by ashb — APPROVED** [review]

**Pull request merged**